centreon/www/include/options/accessLists/reloadACL/reloadACL.php

```diff
@@ -42,69 +42,85 @@
 require_once "./class/centreonMsg.class.php";
 
 if (isset($_GET["o"]) && $_GET["o"] === 'r') {
-    $sid = session_id();
-    $pearDB->query("UPDATE session SET update_acl = '1' WHERE session_id = '" . $pearDB->escape($sid) . "'");
-    $pearDB->query("UPDATE acl_resources SET changed = '1'");
     $msg = new CentreonMsg();
-    $msg->setTextStyle("bold");
-    $msg->setText(_("ACL reloaded"));
-    $msg->setTimeOut("3");
-    passthru(_CENTREON_PATH_ . "/cron/centAcl.php");
+    $sid = session_id();
+    try {
+        $statement = $pearDB->prepareQuery(
+            <<<SQL
+                UPDATE session SET update_acl = '1' WHERE session_id = :sessionId
+                SQL
+        );
+        $pearDB->executePreparedQuery($statement, ['sessionId' => $sid]);
+        $pearDB->executeQuery(
+            <<<SQL
+                UPDATE acl_resources SET changed = '1'
+                SQL
+        );
+        $msg->info(_("ACL reloaded"));
+        passthru(_CENTREON_PATH_ . "/cron/centAcl.php");
+    } catch (CentreonDbException|\PDOException $e) {
+        $msg->error(_("An error occurred"));
+    }
 } elseif (isset($_POST["o"]) && $_POST["o"] === 'u') {
-    $sel = filter_var_array(
+    $msg = new CentreonMsg();
+    $userIds = filter_var_array(
```

Comment:
You can add array_filter directly.
```suggestion
    $userIds = array_filter(filter_var_array(
```

Reply:
I prefer not to as filter_var_array can also return null or false

Reply:
but indeed i didnt check l80 that userIds is actually an array 👀

Reply:
You can do this:
```suggestion
    $userIds = array_filter(filter_var_array(...) ?: []);
```